src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py

```diff
@@ -56,18 +56,25 @@ def _get_detail_error(self, ex):
                               "azure/app-service-web/app-service-linux-intro")
                 elif 'Not enough available reserved instance servers to satisfy' in detail:
                     detail = ("Plan with Linux worker can only be created in a group " +
-                              "which has never contained a Windows worker. Please use " +
-                              "a new resource group. Original error:" + detail)
+                              "which has never contained a Windows worker, and vice versa. " +
+                              "Please use a new resource group. Original error:" + detail)
             return CLIError(detail)
         except: #pylint: disable=bare-except
             return ex
 
-def create_webapp(resource_group_name, name, plan):
+def create_webapp(resource_group_name, name,
+                  plan=None, create_plan=False,
+                  sku=None, is_linux=False, number_of_workers=None,
+                  location=None):
     client = web_client_factory()
-    if is_valid_resource_id(plan):
-        plan = parse_resource_id(plan)['name']
-    location = _get_location_from_app_service_plan(client, resource_group_name, plan)
-    webapp_def = Site(server_farm_id=plan, location=location)
+    plan_id = plan
+    if create_plan:
+        logger.warning("Create appservice plan: '%s'", plan)
```

Comment:
Why are we issuing a warning?  We don't do this for other creates (VM/VMSS/LB/AG) that create extra resources.